src/app/screens/ConfirmPayment/index.tsx

```diff
@@ -49,6 +50,9 @@ function ConfirmPayment(props: Props) {
   );
   const [fiatAmount, setFiatAmount] = useState("");
 
+  const isOriginExtension =
+    originRef?.current?.metaData?.url?.indexOf("chrome-extension") !== -1;
```

Comment:
this looks a lot like a hack. 
and I don't expect this to work in other browsers.

if we don't have the origin we should also not pass it in. 
right now this is only done because the screens and actions need it. 
but ideally this dependency is not there - or what do you think? 

such a hack would need to be replicated in all the places.